File: Insert GCDs in Linked List - Leetcode 2807/Insert GCDs in Linked List - Leetcode 2807.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var insertGreatestCommonDivisors = function(head) {
    let current = head
    while(current.next){
        let next = current.next
        let hcf = gcd(current.val,next.val)
        let node = new ListNode(hcf,next)
        current.next = node;
        current = next
    }
    return head
};

let gcd = function(num1,num2){
    let smaller = Math.min(num1,num2)
    let hcf = 1
    for(let i = 1 ;i<=smaller;i++){
        if(num1%i == 0 && num2%i == 0) hcf = i
    }
    return hcf
}

File: Insert GCDs in Linked List - Leetcode 2807/Insert GCDs in Linked List - Leetcode 2807.java
import java.math.BigInteger;

public class Solution {
    public ListNode insertGreatestCommonDivisors(ListNode head) {
        ListNode prev = head;
        ListNode cur = head.next;

        while (cur != null) {
            int gcd = gcd(cur.val, prev.val); // A
            ListNode g = new ListNode(gcd);
            prev.next = g;
            g.next = cur;
            prev = cur;
            cur = cur.next;
        }

        return head;
    }

    private int gcd(int a, int b) {
        return BigInteger.valueOf(a).gcd(BigInteger.valueOf(b)).intValue();
    }
}
